### PR TITLE
:book: Update p2p docs, general enhancements

### PR DESCRIPTION
**Pull request by mudler**
**What this PR does / why we need it**: It includes docs for #2 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by mudler** [comment]
This needs to be merged with 1.4, as will include changes into the p2p/HA stack of Kairos

File: docs/content/en/docs/Installation/p2p.md [hunk 1]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
 
 {{% /alert %}}
 
-This section will guide you on how to leverage the peer-to-peer (P2P), full-mesh capabilities of Kairos.
+Deploying Kubernetes at the Edge can be a complex and time-consuming process, especially when it comes to setting up and managing multiple clusters. To make this process easier, Kairos leverages peer-to-peer technology to automatically coordinate and create Kubernetes clusters without the need for a control management interface.
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:17]:
```suggestion
Deploying Kubernetes at the Edge can be a complex and time-consuming process, especially when it comes to setting up and managing multiple clusters. To make this process easier, Kairos leverages peer-to-peer technology to automatically coordinate and create Kubernetes clusters without the need of a control management interface.
```

File: docs/content/en/docs/Installation/p2p.md [hunk 2]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
 
-Kairos supports P2P full-mesh out of the box. That allows to seamlessly interconnect clusters and nodes from different regions into a unified overlay network. Additionally, the same network is used for coordinating nodes automatically allowing self-automated, node bootstrap.
+With this feature, users can simply avoid specifying any setting, or just set the number of master nodes in a HA cluster they want to create and the necessary configuration details, and Kairos will take care of the rest. The peer-to-peer technology allows the nodes in the cluster to communicate and coordinate with each other, ensuring that the clusters are set up correctly and efficiently with K3s.
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:19]:
```suggestion
With this feature, users don't need to specify any network settings. They can just set the desired number of master nodes (in the case of an HA cluster) and the necessary configuration details, and Kairos will take care of the rest. The peer-to-peer technology allows the nodes in the cluster to communicate and coordinate with each other, ensuring that the clusters are set up correctly and efficiently with K3s.
```

File: docs/content/en/docs/Installation/p2p.md [hunk 3]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
 
-A hybrid network is automatically set up between all the nodes, so there is no need to expose them over the Internet or expose the Kubernetes management API outside, reducing the attacker's exploiting surface.
+This not only saves users time and effort, but also makes it easier to deploy and manage Kubernetes clusters at the Edge, allowing users to focus on running and scaling their applications.
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:21]:
```suggestion
This makes it easier to deploy and manage Kubernetes clusters at the Edge, saving user's time and effort, allowing them to focus on running and scaling their applications.
```

File: docs/content/en/docs/Installation/p2p.md [hunk 4]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
 
-Kairos can be configured to automatically bootstrap a Kubernetes cluster with the full-mesh functionalities, or it can include an additional interface to the machines to let them communicate within a new network segment.
+This feature is currently experimental and can be optionally enabled by adding the following configuration to the node deployment file, if you are not familiar with the installation process, it is suggested to follow the [quickstart](/docs/getting-started):
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:23]:
```suggestion
This feature is currently experimental and can be optionally enabled by adding the following configuration to the node deployment file. If you are not familiar with the installation process, it is suggested to follow the [quickstart](/docs/getting-started):
```

File: docs/content/en/docs/Installation/p2p.md [hunk 5]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
+
+```yaml
+p2p:
+  # Disabling DHT makes co-ordination to discover nodes only in the local network
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:27]:
```suggestion
 # Disabling DHT makes co-ordination to discover nodes only in the local network
```

File: docs/content/en/docs/Installation/p2p.md [hunk 6]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
+ disable_dht: true #Enabled by default
+ # Automatic cluster deployment configuration
+ auto:
+   ha:
+     # Enables HA controlplane
+     enable: true
+     # number of HA master node (beside the one used for init) for the control-plane
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:34]: I think in the code we ended up using this as the total number of nodes. So in this example, there will be 2 master nodes once of which will be used for init. Is that correct?

File: docs/content/en/docs/Installation/p2p.md [hunk 7]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
+- name: "kairos"
+  passwd: "kairos"
+  ssh_authorized_keys:
+  - github:mudler
+p2p:
+ network_token: "YOUR_TOKEN_GOES_HERE"
 ```
 
-The `kairos` block is used to configure settings to the mesh functionalities. The minimum required argument is the `network_token` and there is no need to configure `k3s` manually with the `k3s` block as it is already implied.
+The `p2p` block is used to configure settings to the mesh functionalities. The minimum required argument is the `network_token` and there is no need to configure `k3s` manually with the `k3s` block as it is already implied.
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:65]:
```suggestion
The `p2p` block is used to configure settings related to the mesh functionalities. The minimum required argument is the `network_token` and there is no need to configure `k3s` manually with the `k3s` block as it is already implied.
```

File: docs/content/en/docs/Installation/p2p.md [hunk 8]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
+| `p2p.role` | Force a specific role for the node of the cluster |
+| `p2p.vpn` | Configures a VPN for the cluster nodes |
+| `p2p.vpn.create` | Enables the creation of the VPN |
+| `p2p.vpn.use` | Enables the use of the VPN for routing Kubernetes traffic of the cluster |
+| `p2p.vpn.env` | Configures the environment variables used to start for the VPN |
+| `p2p.vpn.auto` | Configures the automatic deployment of the cluster |
+| `p2p.auto.enable` | Enables automatic node configuration for role assignment |
+| `p2p.auto.ha` | Configures the high availability settings for the cluster |
+| `p2p.auto.ha.enable` | Enables the high availability settings |
+| `p2p.auto.ha.master_nodes` | The number of expected HA master nodes in the cluster besides the cluster-init master. |
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:153]: same here, is it the total number?

File: docs/content/en/docs/Installation/p2p.md [hunk 9]
```diff
@@ -14,68 +14,215 @@ Feedback and bug reports are welcome, as we are improving the p2p aspects of Kai
...
+  enable: true
+  # Specifies a KubeVIP Interface
+  interface: "ens18"
+```
+
+
+| Keyword | Description |
+| --- | --- |
+| `kubevip` | Block to configure KubeVIP for the cluster |
+| `kubevip.eip` | The Elastic IP used for KubeVIP. Specifying one automatically enables KubeVIP |
```
Comment (jimmykarily) [inline docs/content/en/docs/Installation/p2p.md:186]: Maybe we should explain that they should pick a free IP address in their network. One that won't collide with any other machine or DHCP.

File: docs/content/en/docs/_index.md [hunk 10]
```diff
@@ -8,72 +8,63 @@ menu:
...
-- Node configuration via a single cloud-init config file.
-- Handles airgap upgrades with in-cluster, container registries.
-- Extends the image in runtime or builds time via Kubernetes Native APIs.
-- Plans to support CAPI with full device lifecycle management.
-- Plans to support up to RKE2, kubeadm, and much more!
-- Nodes can optionally connect autonomously via a full-mesh P2P hybrid VPN network. It allows to stretch a cluster up to 10000 km!
-  Kairos can create private virtual network segments to enhance your cluster perimeter without any SPOF.
+- Build your own cloud on-premises with complete control and no vendor lock-in
+- Provision nodes with your own image or use Kairos releases for added flexibility
+- Use Kairos for a wide range of use cases, from Kubernetes applications to appliances and more
```
Comment (jimmykarily) [inline docs/content/en/docs/_index.md:39]: Maybe add:

```
- Simple and streamlined day-2 operations (e.g. node upgrades)
```

File: docs/content/en/docs/_index.md [hunk 11]
```diff
@@ -8,72 +8,63 @@ menu:
...
-Kairos treats all the OSes homogeneously in a distro-agnostic fashion.
+## Features
+
+- Easily create multi-node Kubernetes clusters with [K3s](https://k3s.io), and enjoy all of [K3s](https://k3s.io)'s features
+- Upgrade manually via CLI or with Kubernetes, and use container registries for distribution upgrades
+- Enjoy the benefits of an immutable distribution that stays configured to your needs
+- Configure nodes with a single cloud-init config file for added simplicity
+- Upgrade even in airgap environments with in-cluster container registries
+- Extend your image at runtime or build time with Kubernetes Native APIs
+- Look forward to support for CAPI with full device lifecycle management, and more
```
Comment (jimmykarily) [inline docs/content/en/docs/_index.md:62]: Move this to the bottom with a `coming soon: ` prefix? E.g.

```
- coming soon: CAPI support with full device lifecycle management and more
```

**Review by jimmykarily — DISMISSED** [review]

File: docs/content/en/docs/_index.md (hunk 10 above)
Comment (mudler) [reply docs/content/en/docs/_index.md:39]: fixed!

**Comment by mudler** [comment]
@check-spelling-bot apply [updates](https://github.com/kairos-io/kairos/actions/runs/3776811325/attempts/1).

**Comment by mudler** [comment]
@check-spelling-bot apply [updates](https://github.com/kairos-io/kairos/actions/runs/3776811325/attempts/1).

**Comment by mudler** [comment]
@check-spelling-bot apply [updates](https://github.com/kairos-io/kairos/actions/runs/3776816103/attempts/1).

**Pull request merged**